cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ReconcilerUtils.java

```diff
@@ -381,4 +381,23 @@ public static boolean nodePoolsEnabled(Kafka kafka) {
     public static boolean kraftEnabled(Kafka kafka) {
         return KafkaCluster.ENABLED_VALUE_STRIMZI_IO_KRAFT.equals(Annotations.stringAnnotation(kafka, Annotations.ANNO_STRIMZI_IO_KRAFT, "disabled").toLowerCase(Locale.ENGLISH));
     }
+
+    /**
+     * Creates a hash from Secret's data.
+     * @param secret Secret with data.
+     * @return Hash of all secret values.
+     */
+    public static String hashSecretContent(Secret secret) {
+        if (secret == null) {
+            throw new RuntimeException("Secret not found");
+        }
+        StringBuilder sb = new StringBuilder();
+        for (String key : secret.getData().keySet()) {
+            sb.append(secret.getData().get(key));
+        }
+        if (sb.toString().length() == 0) {
```

Comment:
what if we do this earlier by checking `secret.getData() != null && secret.getData().size() != 0` instead of creating a new `StringBuilder` (even if empty) which could not be needed at all? Just a suggestion.

Reply:
Makes sense thanks.